packages/react/src/authentication/hooks/useSocialLogin.ts

```diff
@@ -0,0 +1,239 @@
+import {
+  type BlackoutError,
+  type Config,
+  postAccountLink,
+  type PostAccountLinkData,
+  postSocialLogin,
+  type PostSocialLoginData,
+} from '@farfetch/blackout-client';
+import { useCallback, useReducer, useRef } from 'react';
+
+const actionTypes = {
```

Comment:
Please separate this logic and place it on the redux package. Make sure to use the correct prefixes on each actionType.